swim/ping_handler.go

```diff
@@ -47,6 +62,7 @@ func handlePing(node *Node, req *ping) (*ping, error) {
 		Changes:           changes,
 		Source:            node.Address(),
 		SourceIncarnation: node.Incarnation(),
+		App:               node.app,
```

Comment:
This is the reason why the integration tests are failing due to a json-schema validation error. You only added `app` to ping-request while also sending in the ping-response here.